src/data/Dhis2Events.ts

```diff
@@ -21,3 +29,19 @@ export async function postEvents(api: D2Api, events: Event[]): Promise<Synchroni
 
     return eventsResult;
 }
+
+function setEnrollmentToEvent(
+    events: Event[],
+    options: { existingTeis: TrackedEntityInstance[]; teis: TrackedEntityInstance[] }
+) {
+    const { existingTeis, teis } = options;
+    return teis.length === 0
+        ? events
+        : events.map(event => {
+              const existingTei = existingTeis.find(tei => tei.id === event.trackedEntity);
+              const tei = teis.find(tei => tei.id === event.trackedEntity);
+              if (!tei) return event;
+              const enrollmentId = existingTei?.enrollment?.id || getUid([tei.id, tei.orgUnit, tei.program].join("-"));
```

Comment:
Regarding the fallback with a join value: we should add a comment here for clarification. Better yet, we should consider using the common function. This likely relates to **`src/data/Dhis2TrackedEntityInstances.ts:368`** where `getUid([tei.id, orgUnit.id, program.id].join("-"))` is used. When the ID cannot be found, we assume it was generated by our app and reuse the same seed for the pseudo-ID derived from that join. 

This raises an important question (which could be added to the comment): **Is this approach entirely safe?** what happens if the UID is different? could this result in **duplicate enrollments**? is that scenario possible?